syntax = "proto3";

package myapp;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

message User {
  int32 id = 1;
  string name = 2;
  bool is_active = 3;
  google.protobuf.Timestamp created_at = 4;
}

message UserRequest {
  int32 id = 1;
}

message UserResponse {
  User user = 1;
}

message UploadSummary {
  int32 count = 1;
}

message ChatMessage {
  string sender = 1;
  string text = 2;
}

service UserService {
  rpc GetUser (UserRequest) returns (UserResponse);
  rpc ListUsers (google.protobuf.Empty) returns (stream User);
  rpc UploadUsers (stream User) returns (UploadSummary);
  rpc Chat (stream ChatMessage) returns (stream ChatMessage);
}
